framework/wazuh/core/batcher/buffer.py

```diff
@@ -0,0 +1,83 @@
+import sys
+from typing import List
+
+from wazuh.core.batcher.mux_demux import Message
+
+
+class Buffer:
+    """
+    Manages the buffer for batching messages.
+
+    Parameters
+    ----------
+    max_elements : int
+        Maximum number of messages in the buffer.
+    max_size : int
+        Maximum size of the buffer in bytes.
+    """
+    def __init__(self, max_elements: int, max_size: int):
+        self.max_elements = max_elements
+        self.max_size = max_size
+        self._buffer: List[Message] = []
+
+    def add_message(self, msg: Message):
```

Comment:
We should check that the buffer size doesn't not trespasses the `max_elements` and `max_size` limits. We shouldn't store new messages if it's already full

Reply:
Changed in https://github.com/wazuh/wazuh/commit/7344dcea68aa1f5d451757e21ea6e02cc5e61448.